pyproj/_geod.pyx

```diff
@@ -161,46 +184,151 @@ cdef class Geod:
         double lat2,
         int npts,
         bint radians=False,
+        int initial_idx=1,
+        int terminus_idx=1,
+        object out_lons=None,
+        object out_lats=None,
     ):
         """
         given initial and terminus lat/lon, find npts intermediate points.
+        returns lons, lats buffers
+        """
+        res = \
+            self._inv_or_fwd_intermediate(
+                lon1=lon1,
+                lat1=lat1,
+                lon2_or_azi1=lon2,
+                lat2_or_nan=lat2,
+                npts=npts,
+                del_s=0,
+                radians=radians,
+                initial_idx=initial_idx,
+                terminus_idx=terminus_idx,
+                flags = GEOD_INTER_FLAG_AZIS_DISCARD,
+                out_lons=out_lons,
+                out_lats=out_lats,
+                out_azis=None,
+            )
+
+        return res.lons, res.lats
+
+    @cython.boundscheck(False)
```

Comment:
Since you allow users to pass in their own arrays, we can no longer disable the boundscheck [ref](https://cython.readthedocs.io/en/latest/src/userguide/numpy_tutorial.html#tuning-indexing-further).

> if you ‘’do’’ happen to access out of bounds you will in the best case crash your program and in the worst case corrupt data

I am wondering what the performance is for the two different scenarios:
1. Passing in numpy array with boundscheck enabled.
2. Returning `array.array` with boundscheck disabled and convert to numpy array later.

Reply:
https://github.com/pyproj4/pyproj/pull/841/files#diff-884caf2af6e10fc8e0b305d38b427edc7b43d9ed3ec739aed1be2de94049ab7dR310

Isn't this check enough?

Reply:
I missed that, thanks for pointing it out. That is good to have. I am still curious about the performance difference between the previous scenarios (minus the boundscheck part).

Reply:
> I missed that, thanks for pointing it out. That is good to have. I am still curious about the performance difference between the previous scenarios (minus the boundscheck part).

That would be interesting to investigate...
My Cython training is just from this PR, I guess I could try to dig deeper later about it after I get more familiar with the language.

Reply:
I've been thinking about your scenarios, I think the best way to put it into a test would be to make different versions of the `inv_or_fwd_intermediate` function, two as you suggested plus my previous version that doesn't use `with gil` for the case (`inv` + `del_s` + no input array).
I think this PR is already mature enough and this last version of `inv_or_fwd_intermediate` function is "one size fits all", with only a potential minor performance loss over my previous version.
I don't want these tests to push the release schedule of v3.1 further so I'm squashing my last commit and will open a new PR after the release of v3.1 to test these performance issues.

Reply:
I did some basic tests and it seems like the performance improves a bit when using numpy arrays. So, probably good to leave it as is.